### DISCUSSION TITLE
Going from AssumeRoleProvider to Credentials

### DISCUSSION THREAD

**Discussion by lcmgh** [awslabs/aws-sdk-rust#814] [Q&A] [answered]
Hi!

To make it much easier to connect to RDS databases I am working on a command line user interface that

* auto detects bastion hosts for RDS clusters (based on tag matching)
* lists rds clusters
* establishes tunnel via aws cli ssm + ssh (and remembers the port it used the last time for a certain connection)

Next I want to make the app generate temporary access tokens for databases that support IAM based authentication and then spawn a db client (psql, dbeaver etc.) that auto-establishes the connection.

For generating the temporary password I am usnig this code: [https://docs.rs/aws-sig-auth/0.55.1/aws_sig_auth/#generate-rds-iam-token ](https://docs.rs/aws-sig-auth/0.55.1/aws_sig_auth/#generate-rds-iam-token)

However I am facing an issue that I have described here: [https://github.com/awslabs/aws-sdk-rust/issues/792#issuecomment-1543798866 ](https://github.com/awslabs/aws-sdk-rust/issues/792#issuecomment-1543798866)

Could you please help regarding that?

Thanks

**Comment by jdisanti** [comment, marked as answer]
`AssumeRoleProvider` implements the `ProvideCredentials` trait, which you can use to resolve the actual credentials:
https://docs.rs/aws-config/0.55.3/aws_config/sts/struct.AssumeRoleProvider.html#impl-ProvideCredentials-for-AssumeRoleProvider

Something like:
```rust
let credentials = provider.provide_credentials().await?;
```
